pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -291,13 +257,132 @@ func (pl *VolumeZone) EventsToRegister() []framework.ClusterEventWithHint {
+			logger.V(10).Info("Cannot schedule pod onto node due to volume (mismatch on label key)", "pod", klog.KObj(pod), "node", klog.KObj(node), "PV", klog.KRef("", pvTopology.pvName), "PVLabelKey", pvTopology.key)
+			return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonConflict), false
+		}
+	}
+	return nil, true
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim added or updated.
+// It checks whether the change of PVC has made a previously unschedulable pod schedulable.
+// A PVC becoming bound or using a WaitForFirstConsumer storageclass can cause the pod to become schedulable.
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPVC, modifiedPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
+	}
+	isMatched := pl.checkPVCBindingToPodPV(logger, modifiedPVC, pod)
+	// updated PVC is not schedulable because PVC doesn't match the pod's PVC
+	if !isMatched {
+		logger.V(5).Info("PVC was created or updated but it doesn't make this pod schedulable.", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+		return framework.QueueSkip, nil
+	}
+	wasMatched := pl.checkPVCBindingToPodPV(logger, originalPVC, pod)
+	// the PVC that didn't match the pod now matches the pod
+	if isMatched && !wasMatched {
+		logger.V(5).Info("PVC was created or updated, which might make the pod schedulable. The given PVC matches the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+		return framework.Queue, nil
+	}
+	logger.V(5).Info("PVC was created or updated but it doesn't make this pod schedulable. Nothing has changed about PV bound to PVC.", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+	return framework.QueueSkip, nil
+}
+
+// checkPVCBindingToPodPV verifies if the PVC is bound to PV of a given Pod.
+func (pl *VolumeZone) checkPVCBindingToPodPV(logger klog.Logger, pvc *v1.PersistentVolumeClaim, pod *v1.Pod) bool {
+	if pvc == nil {
+		return false
+	}
+	pvcNames := pl.getPersistentVolumeClaimNameFromPod(pod)
+	for _, pvcName := range pvcNames {
+		if (pvc.Name != pvcName) || (pod.Namespace != pvc.Namespace) {
+			// pod's PVC doesn't match with the given PVC
+			continue
+		}
+		logger.V(5).Info("PVC matches the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(pvc))
+		pvName := pvc.Spec.VolumeName
+		if pvName == "" {
+			scName := storagehelpers.GetPersistentVolumeClaimClass(pvc)
+			if len(scName) == 0 {
+				return false
+			}
+			isWait, _ := pl.isWaitForFirstConsumer(scName)
+			if !isWait {
+				logger.V(5).Info("PVC is bound to storageClass but the volumeBindingMode is not WaitForFirstConsumer", "storageClass", scName)
+				return false
+			}
+		}
+		return true
```

Comment:
Let's make the implementation around here simpler for now. As you may know, there's a concern of throughput of QHints.
So, here, as a first step, we should make it minimum.
And, after we establish some observability around QHint(https://github.com/kubernetes/kubernetes/issues/124566), we may be able to have this detailed filtering if we could ensure that wouldn't impact a large cluster negatively.

```suggestion
		return true
```

Reply:
> Even though this logic is already implemented here, do you want to remove it?

Yes. We don't need perfect filtering on QHint. 

Rather if we pursued perfection, it could put additional cost for throughput for a little gain.
As you know, the throughput of QHints is one of the biggest hurdles we're facing right now. That's why I said -
> as a first step, we should make it minimum.
And, after we establish some observability around QHint(https://github.com/kubernetes/kubernetes/issues/124566), we may be able to have this detailed filtering if we could ensure that wouldn't impact a large cluster negatively.

Reply:
Or did you mean if we go with my suggestion, we could mistakenly return QueueSkip in some scenarios making Pod schedulable?

Reply:
@sanposhiho 

Ah previously I had misunderstood that this function is also used within Filter function, but I understood that this function is only used within QHint logic.

> Or did you mean if we go with my suggestion, we could mistakenly return QueueSkip in some scenarios making Pod schedulable?

However, I think it is possible to return QueueSkip mistakenly, like just updating the storageClass of PVC, or updating the PV name bound to PVC etc.
I think current implementation is better in terms of safety.
But, if you really think that simpler logic is better than safer logic, I am fine. I will follow your comment.

Reply:
I pursue simplicity here though, we cannot go to the option where we may return Skip mistakenly. (that's an unacceptable bug)

What if we eliminate `wasMatched` check in isSchedulableAfterPersistentVolumeClaimChange? 
I believe then we can apply my suggestion (eliminating `GetPersistentVolumeClaimClass` here)?

```
func (pl *VolumeZone) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
	originalPVC, modifiedPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
	if err != nil {
		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
	}

	isMatched := pl.checkPVCBindingToPodPV(logger, modifiedPVC, pod)
	// updated PVC is not schedulable because PVC doesn't match the pod's PVC
	if isMatched {
		return framework.Queue, nil
	}

	return framework.QueueSkip, nil
}
```

Reply:
Thanks, SGTM. I've updated it based on comments.